conan/internal/cache/db/cache_database.py

```diff
@@ -90,9 +90,10 @@ def create_recipe(self, path, ref: RecipeReference):
     def create_package(self, path, ref: PkgReference, build_id):
         self._packages.create(path, ref, build_id=build_id)
 
-    def list_references(self):
+    def list_references(self, pattern=None):
         return [d["ref"]
-                for d in self._recipes.all_references()]
+                for d in self._recipes.all_references()
+                if pattern is None or d["ref"].partial_match(pattern)]
```

Comment:
We could further move this check to the SQL query instead of here, but it seems too much effort for little gain to port the `partial_match` functionality to the sdk

Reply:
Yes, good for me here.